bionic/cache.py

```diff
@@ -1028,7 +1041,7 @@ def update_hash(hash_, obj):
 
 def relativize_url(artifact_url, metadata_url):
```

Comment:
To avoid the kind of bug we had before, maybe it would be safer to only perform path operations on path objects, not on URLs? I.e., convert the input URLs to paths, perform the transformation, then convert the output back to a URL?